### PR TITLE
[libc++] Replace 'tags' in CSV status pages by inline notes

### PR DESCRIPTION
**Pull request by ldionne**
This patch replaces 'tags' in the CSV status pages by inline notes
that optionally describe more details about the paper/LWG issue.

Tags were not really useful anymore because we have a vastly superior
tagging system via Github issues, and keeping the tags up-to-date
between CSV files and Github is going to be really challenging.

This patch also adds support for encoding custom notes in the CSV
files via Github issues. To encode a note in the CSV file, the
body (initial description) of a Github issue can be edited to contain
the following markers:

    BEGIN-RST-NOTES
    text that will be added as a note in the RST
    END-RST-NOTES

Amongst other things, this solves the problem of conveying that a
paper has been implemented as a DR, and it gives a unified way to
add notes to the status pages from Github.

### CODE REVIEW

File: libcxx/docs/Status/Cxx2cPapers.csv [hunk 1]
```diff
@@ -1,78 +1,78 @@
...
-"`P2836R1 <https://wg21.link/P2836R1>`__","``std::basic_const_iterator`` should follow its underlying type's convertibility","2023-11 (Kona)","","","|DR|"
-"`P2264R7 <https://wg21.link/P2264R7>`__","Make ``assert()`` macro user friendly for C and C++","2023-11 (Kona)","","",""
-"`P1673R13 <https://wg21.link/P1673R13>`__","A free function linear algebra interface based on the BLAS","2023-11 (Kona)","","",""
-"","","","","",""
-"`P2875R4 <https://wg21.link/P2875R4>`__","Undeprecate ``polymorphic_allocator::destroy`` for C++26","2024-03 (Tokyo)","|Complete|","15.0",""
-"`P2867R2 <https://wg21.link/P2867R2>`__","Remove Deprecated ``strstreams`` From C++26","2024-03 (Tokyo)","|Complete|","19.0",""
-"`P2869R4 <https://wg21.link/P2869R4>`__","Remove Deprecated ``shared_ptr`` Atomic Access APIs from C++26","2024-03 (Tokyo)","","",""
-"`P2872R3 <https://wg21.link/P2872R3>`__","Remove ``wstring_convert`` From C++26","2024-03 (Tokyo)","|Complete|","19.0",""
-"`P3107R5 <https://wg21.link/P3107R5>`__","Permit an efficient implementation of ``std::print``","2024-03 (Tokyo)","","","|format| |DR|"
```
Comment (ldionne) [inline libcxx/docs/Status/Cxx2cPapers.csv:53]: I guess we want to retain a way to annotate things in the CSV as being DRs, though.

**Comment by ldionne** [comment]
I tested this locally and verified that the output was reasonable. I'm going to merge this without waiting because of the massive potential for merge conflicts, and resolving such conflicts would be very challenging from this side.

**Comment by philnik777** [comment]
I have to say that IMO this significantly decreased the readability of the status pages. Most of the page is just white space now and the paper names are super squished and almost always in multiple lines.

**Comment by ldionne** [comment]
@philnik777 I agree. I think the output can be much improved, for example we could probably generate a tooltip instead. I'll look into that.

**Pull request merged**